### PR TITLE
OCPBUGS-29804: Fix crash if helm chart metadata is nil

### PR DESCRIPTION
**Pull request by jerolimov**
**Fixes:**
Fixes an internal crash related to https://issues.redhat.com/browse/OCPBUGS-29804 without updating the helm library.

Related to #13816

I added just one new nil check:

```go
if entries[i] == nil || entries[i].Metadata == nil {
  klog.Warningf("Helm chart %v from repository %v has an invalid entry at index %v", key, helmRepo.Name, i)
  entries = append(entries[:i], entries[i+1:]...)
  continue
}
```

The rest of the change is only because I renamed the array `entry` to `entries`.

**Test setup:**
1. Open the console and switch to the developer perspective
2. Navigate to Helm
3. Add this helm repository: `https://jerolimov.github.io/helm-cve-2024-26147-test/`
   * Repo source code: https://github.com/jerolimov/helm-cve-2024-26147-test
   * Broken repository: https://github.com/jerolimov/helm-cve-2024-26147-test/commit/f96fc8247455dbc96550f415f32722163f5cd1ad
4. Click Create > Helm Release to search for available helm charts.

Previously the page showed an error.

With this change the page works fine also with this broken helm repository.

### CODE REVIEW

File: pkg/helm/chartproxy/proxy.go [hunk 1]
```diff
@@ -93,29 +93,34 @@ func (p *proxy) IndexFile(onlyCompatible bool, namespace string) (*repo.IndexFil
...
-				for i := len(entry) - 1; i >= 0; i-- {
-					if entry[i].Type == "library" {
-						entry = append(entry[:i], entry[i+1:]...)
+			for key, entries := range idxFile.Entries {
+				for i := len(entries) - 1; i >= 0; i-- {
+					if entries[i] == nil || entries[i].Metadata == nil {
+						klog.Warningf("Helm chart %v from repository %v has an invalid entry at index %v", key, helmRepo.Name, i)
+						entries = append(entries[:i], entries[i+1:]...)
 						continue
 					}
```
Comment (jerolimov) [inline pkg/helm/chartproxy/proxy.go:102]: This is actually the 5 new lines.

The rest is only because I renamed entry to entries because it's an array.

Comment (jerolimov) [reply pkg/helm/chartproxy/proxy.go:102]: /cc @Kartikey-star @jhadvig

**Review by Kartikey-star — APPROVED** [review]
/lgtm

**Pull request merged**